### PR TITLE
IEP-952: Espressif IDE popup for old tool versions popping up after installing tools via IDE

### PR DESCRIPTION
**Pull request by alirana01**
## Description

Checking the esp_idf.json timestamp for env initialization

Fixes # ([IEP-952](https://jira.espressif.com:8443/browse/IEP-952))

## Type of change

Please delete options that are not relevant.
- Bug fix (non-breaking change which fixes an issue)

## How has this been tested?

Use the tests in the following PR (https://github.com/espressif/idf-eclipse-plugin/pull/662). 
Also try to install the tools in the IDE after the IDE is initialized via esp_idf.json file and try to close and reopen the IDE to see if the message popup appears. The original jira ticket also contains links to the forum and github issues.

**Test Configuration**:
* ESP-IDF Version: Any
* OS (Windows,Linux and macOS): Windows only (Since installer is currently given for Windows only)

## Checklist
- [ ] PR Self Reviewed
- [ ] Applied Code formatting

### CODE REVIEW

File: bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/InitializeToolsStartup.java [hunk 1]
```diff
@@ -56,10 +58,8 @@ public class InitializeToolsStartup implements IStartup
 	private static final String IDF_INSTALLED_LIST_KEY = "idfInstalled"; //$NON-NLS-1$
 	private static final String PYTHON_PATH = "python"; //$NON-NLS-1$
 	private static final String IDF_PATH = "path"; //$NON-NLS-1$
-	private static final String IS_INSTALLER_CONFIG_SET = "isInstallerConfigSet"; //$NON-NLS-1$
 	private static final String DOC_URL = "\"https://docs.espressif.com/projects/esp-idf/en/latest/esp32/api-guides/partition-tables.html?highlight=partitions%20csv#creating-custom-tables\""; //$NON-NLS-1$
-
-	private String newIdfPath;
+	private static final String LAST_MODIFIED_ESP_IDF_JSON_FILE = "lastModifed-esp_idf.json"; //$NON-NLS-1$
```
Comment (kolipakakondal) [inline bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/InitializeToolsStartup.java:62]: Name is bit misleading, my initial impression was you're trying to store lastmodified timestamp in the `lastModifed-esp_idf.json` file. How about something like this  `esp_idf_json_lastmodified_timestamp` or similar?

**Review by kolipakakondal — CHANGES_REQUESTED** [review]
LGTM. But this breaks the backward compatability as every user will be prompted with message dialog on IDE restart for the first time (lastmodified > -1)

Can we do something about it?

File: bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/InitializeToolsStartup.java [hunk 2]
```diff
@@ -99,27 +99,19 @@ public void run()
 			Logger.log(MessageFormat.format("esp-idf.json file doesn't exist at this location: '{0}'", url.getPath()));
 			return;
 		}
-		else if (isInstallerConfigSet())
+		else if (isConfigFileUpdated(idf_json_file))
```
Comment (sigmaaa) [inline bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/InitializeToolsStartup.java:102]: Hi @alirana01, what will happen, if idf_json_file is modified and we click `NO`? It looks like if we store preferences like that: 
`prefs.putLong(LAST_MODIFIED_ESP_IDF_JSON_FILE, idf_json_file.lastModified());` then this condition returns false after second eclipse restart and we will update our environment with new idf_json_file anyway `updateEnvUsingIdfJsonFile(idf_json_file);`. It's not clear to me, to be honest, can you explain?

File: bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/InitializeToolsStartup.java [hunk 3]
```diff
@@ -134,7 +126,11 @@ else if (isInstallerConfigSet())
 			});
 		}
 
-		// read esp-idf.json file
+		updateEnvUsingIdfJsonFile(idf_json_file);
```
Comment (sigmaaa) [inline bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/InitializeToolsStartup.java:129]: even if we click No, we will get here after the second eclipse restart, am I missing something here?

**Review by sigmaaa — COMMENTED** [review]
Hi @alirana01, some moments are not clear to me, can you explain, please?

**Comment by alirana01** [comment]
I have added the required changes @kolipakakondal @sigmaaa kindly review again

**Review by sigmaaa — APPROVED** [review]
LGTM

**Review by kolipakakondal — APPROVED** [review]
LGTM

**Comment by AndriiFilippov** [comment]
@alirana01 hi ! 

Tested: 
OS - Windows 10 
ESP-IDF: v4.4 / v5.2 

Install tools v4.4 via Windows Tools Installer -> switch to ESP-IDF v5.2 -> able to see popup window -> "Yes" -> it change the idf path under  esp_idf.json file -> "No" -> update esp_idf.json. 

Do not see popup window again after re-open project.

LGTM 👍

**Pull request merged**